tests/testthat/test-frgtracker.R

```diff
@@ -148,9 +148,91 @@ test_that("The output of rank_courses is not valid", {
 
 #tests for rank_courses end
 
-#function5 start
+# tests for rank_students start
+
+generate_fake_dataframe <- function() {
+  df <- data.frame(
+    course_id = c(rep("511", 4), rep("522", 4)),
+    student_id = c(
+      "tom", "tiff", "mike", "joel",
+      "tom", "tiff", "mike", "joel"
+    ),
+    grade = c(90, 70, 60, 50, 95, 91, 85, 80)
+  )
+  df
+}
+
+generate_fake_dataframe_outputs <- function(courseid, grade) {
```

Comment:
it seems this parameter is not used at all?

Reply:
Yes! sorry its not used. will remove it.